### PR TITLE
BL-955-bd-auth-0-research-prototyping-spike

### PR DESCRIPTION
**Pull request by ashtilawat**
## Description

1. Created a SecurityConfig class with filterChain method to authenticate jwt passed frontend/client. 
2. Created a AudienceValidator class to check that the tokens audience is the same as the application properties audience environmental variable. 
3. Stubbed out User model, UserService, and UserController for future use.

## Loom Video

[Daniel Loom](https://www.loom.com/share/812a1d277d2646fd9c85cb67071664ae)
[Abe Loom](https://www.loom.com/share/842906673f7140d896707eb3812e2225)
[Tyler Loom](https://www.loom.com/share/045130a406474c3fb058ac8344357a34)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,9 +19,13 @@ repositories {
 }
 
 dependencies {
-	implementation 'org.springframework.boot:spring-boot-starter-web'
+	implementation 'org.springframework.security.oauth:spring-security-oauth2:2.5.1.RELEASE'
+    implementation 'org.springframework.boot:spring-boot-starter-oauth2-resource-server'
```
Comment (deepalisethia) [inline build.gradle]: Check indentation on line 23 so that it matches with the rest

**Review by deepalisethia — CHANGES_REQUESTED** [review]
Looks good to me - I would just check the indentations.

**Review by razborovdb — DISMISSED** [review]
Looks good

**Review by razborovdb — DISMISSED** [review]

**Review by razborovdb — APPROVED** [review]

**Review by deepalisethia — APPROVED** [review]
Looks good to me. I don't see anything that stands out after the previous changes were implemented.

**Comment by confect1ondev** [comment]
I assume there are plans to use `EnableWebSecurity.java`, but I'd double check that it isn't bloat from a scrapped implementation. Other than that, it looks good to me.

**Review by confect1ondev — APPROVED** [review]
It looks good, great job!

**Review by dpearson00 — APPROVED** [review]
Everything looks good

**Pull request merged**